tempesta_fw/http.c

```diff
@@ -615,35 +634,212 @@ tfw_http_adjust_resp(TfwHttpResp *resp, TfwHttpReq *req)
 	return tfw_http_set_hdr_connection(m, conn_flg);
 }
 
+static inline bool
+tfw_http_req_is_idempotent(TfwHttpReq *req)
+{
+	return (req->method != TFW_HTTP_METH_POST);
+}
+
 /*
- * Depending on results of processing of a request, either send the request
- * to an appropriate server, or return the cached response. If none of that
- * can be done for any reason, return HTTP 404 or 500 error to the client.
+ * Forward request @req to server connection @srv_conn.
+ * Can be called in both SoftIRQ and kernel thread contexts.
+ */
+static void
+tfw_http_conn_req_forward(TfwConnection *srv_conn, TfwHttpReq *req)
```

Comment:
I'd prefer shorter suffix `fwd` for the group of the function - it's common for forwarding logic and much shorter.